src/main/thrift/parquet.thrift

```diff
@@ -197,18 +197,78 @@ enum FieldRepetitionType {
   REPEATED = 2;
 }
 
+/** Identifier for the sort order used to produce min and max values. */
+enum Order {
+  /**
+   * The signed ordering is the order produced by comparing single primitive
+   * values with a signed comparator, or the lexicographic ordering produced by
+   * comparing each byte of a binary or fixed using a signed comparator.
+   *
+   * (A signed comparator uses the most-significant bit as a sign bit; an
+   * unsigned comparator uses the most-significant bit as part of the value's
+   * magnitude. Note that unsigned comparison is not defined for floating
+   * point values.)
+   */
+  SIGNED = 0;
+
+  /**
+   * The unsigned ordering is produced by comparing single primitive values
+   * with an unsigned comparison, or the lexicographic ordering produced by
+   * comparing eacy byte of a binary or fixed using an unsigned comparator.
+   *
+   * (A signed comparator uses the most-significant bit as a sign bit; an
+   * unsigned comparator uses the most-significant bit as part of the value's
+   * magnitude. Note that unsigned comparison is not defined for floating
+   * point values.)
+   */
+  UNSIGNED = 1;
+
+  /**
+   * Identifiers for custom orderings, to be defined using CustomOrder entries
+   * in the file footer.
+   */
+  CUSTOM1 = 2;
+  CUSTOM2 = 3;
+  CUSTOM3 = 4;
+  CUSTOM4 = 5;
+  CUSTOM5 = 6;
+  CUSTOM6 = 7;
```

Comment:
The problem this enum is trying to solve is to avoid writing collation identifier strings in the metadata of each page. The approach I took was to keep that field small with an enum and translate the custom symbols to collating sequence identifier strings using metadata in the footer.

If we used the union-style struct approach, then "signed", "unsigned", or the identifier string would be present in every page. At a minimum, I think signed and unsigned should be symbols and only custom strings would be in every page.